### ISSUE TITLE
gpAjax.ajaxPost()

### ISSUE THREAD

**Issue by JustAnotherNoobCoder** [Sau1707/Grepolis#3]
Hey. I'm trying to learn more about programming from your project. I was wondering if you could explain what _gpAjax.ajaxPost()_ does? I've done much google searching and have been unable to find anything about this function.

**Comment by Sau1707** [comment]
Hi,
It's a specific function built for grepolis, 

Most of the communication between client and server it's done using http request. You fetch a url with some parameters and the server that receive the request read the parameters and give an answer. 

How the parameters are structured are decided from the developer that build the application, in the case of grepolis every request has:

- a controller
- an action
- some action

Furthermore with every request some extra data has to be passed, like the authentication cookie or the cross origin token for safety reasons. You don't have to worry about passing this data since it's done automatically

This is the implementation of the function (screenshot taken from the game cache)
![ajax](https://user-images.githubusercontent.com/83017316/208293411-7d7fd763-eba1-41ba-b952-9a3d45f0edb4.PNG)

- show_loader I think it's the grepolis spinning while it get the data (never used really)
-  callback takes a function that get called when the request is done

Note:
- ajaxPost it's used for tell the server to do some actions (like send an attack)
- ajaxGet it's used to call for some information (like ask how many troops a city have)

**Comment by JustAnotherNoobCoder** [comment]
Hey, thanks for the detailed response. Although, most of it is wasted on me because I am new to programming. But I understand some of it 👍 

How did you come to find out which functions are being used on Grepolis' backend server?

I would like to eventually start writing my own scripts for grepolis like you. What would you suggest a novice to learn so that he may start soon? I am currently learning html, css and javascript.

**Comment by Sau1707** [comment]
You cannot know what functions are used in the backed, since it's not open source.
What you can do is perform some game actions and see what request it's sent to the server. Then just replicate it.

I'm gonna send you a video on how to do that this evening or tomorrow.

**Comment by JustAnotherNoobCoder** [comment]
Sounds good. Thank you.  

Also why does your bot have no UI? I think that would make it much better

**Comment by Sau1707** [comment]
Because it's a personal project and i don't need it ;)

**Comment by JustAnotherNoobCoder** [comment]
I see you also made an auto boot camp :D 

I've been trying for the longest to do this also. But everytime I tried to use .click() on the element, it would never open for some reason.

You are the best

**Comment by JustAnotherNoobCoder** [comment]
I'm talking about this line by the way. `document.getElementsByClassName("attack_spot attack_possible")[0]`. Whenever I had a .click() on the end to open it, it never opens

**Comment by Sau1707** [comment]
Hi and Happy new year,

I took a lot about it and I'm not sure why it's not working, probably the click event it's bind to the map and not the spot itself.

But what I found:
If you wanna open the attack spot just do
```js
require("features/attack_spots/factories/attack_spot").openWindow()
```
It will automatically open the reward window or the attack one, like if you click the spot

To check if it is the reward window or the the normal one:
```js
MM.getModelByNameAndPlayerId('PlayerAttackSpot').hasReward()
```

And if you wanna have some informations about the attack spot just do
```js
MM.getModelByNameAndPlayerId('PlayerAttackSpot').attributes
```
You can find the level, reward, island, town, ecc...

Hope that this helps

**Comment by JustAnotherNoobCoder** [comment]
Brillaint. Works perfectly. Thanks!
